hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlJsonTestSupport.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastJsonValue;
+import com.hazelcast.map.IMap;
+import com.hazelcast.sql.SqlColumnMetadata;
+import com.hazelcast.sql.SqlColumnType;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.SqlRowMetadata;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class SqlJsonTestSupport extends SqlTestSupport {
+    protected static final ObjectMapper SERIALIZER = new ObjectMapper();
+
+    protected void initComplexObject() {
+        final IMap<Long, HazelcastJsonValue> test = instance().getMap("test");
+        test.put(1L, new HazelcastJsonValue("["
+                + "[1,\"2\",3,{\"t\":1}],"
+                + "{\"t\":1},"
+                + "3"
+                + "]"));
+    }
+
+    protected Object querySingleValue(final String sql) {
+        final Map<String, Object> result = querySingleRow(sql);
+        return result.values().iterator().next();
+    }
+
+    protected Map<String, Object> querySingleRow(final String sql) {
+        return query(sql).get(0);
```

Comment:
Shouldn't we also assert that there are no more rows in the result? Also in `querySingleValue`, that there are no more values in the map?